test/server/controller/plugin.test.ts

```diff
@@ -0,0 +1,134 @@
+import pg from "pg";
+import { Trade } from "../../../src/server/types";
+import Controller from "./../../../src/server/controller/plugin";
+
+describe("Controllers Test", function () {
+  jest.mock("pg");
+
+  const controller = new Controller();
+  const trade: Trade = {
+    id: "abc",
+    ticker: "ABC",
+    company_name: "Test",
+    reference_number: "",
+    unit_price: 0,
+    quantity: 0,
+    total_cost: 0,
+    trade_type: "long",
+    note: "",
+    created_at: "10/10/2020",
+    trade_status: true,
+  };
+
+  describe("Write Action", function () {
+    beforeEach(function () {
+      pg.Client.prototype.query = jest.fn().mockResolvedValue({
+        rows: [
+          {
+            id: "abc",
+          },
+        ],
+      });
+    });
+
+    afterEach(function () {
+      jest.resetAllMocks();
+    });
+
+    it("should write a trade to db and return an id", async function () {
+      const result: {
+        [rows: string]: [{ [id: string]: "" }];
+      } = await controller.write(trade);
+      expect(result.rows[0].id).toEqual("abc");
+      expect(
+        pg.Client.prototype.query
```

Comment:
**Comment**: By electing to stub `pg.Client.prototype.query`, this is no longer a unit test, but an integration test. We should investigate the viability of stubbing `PostgresGateway` instead.

Reply:
Updated all tests to stub controller instead.